rustls/tests/api.rs

```diff
@@ -47,6 +47,216 @@ use provider::cipher_suite;
 use provider::sign::RsaSigningKey;
 use rustls::ProtocolVersion::TLSv1_2;
 
+mod test_rpk {
+    use super::*;
+
+    #[test]
+    fn test_certificate_type_extension_errors() {
+        let kt = KeyType::Rsa2048;
+        //TODO this does not work for TLS13 yet because we don't encrypt the extensions and get the following error: Err(PeerMisbehaved(UnexpectedCleartextExtension))
```

Comment:
style nit: I think the comments in this file should be written `//<space><content>` instead of `//<content>`